viewing_party/party.py

```diff
@@ -1,23 +1,174 @@
+    return user_data
+
+def watch_movie(user_data, title):
+    # for movies["title"] in user_data["watched"] == title:
+    for movies in user_data["watchlist"]:
+        if movies["title"] == title:
+            print(title)
+            user_data["watched"].append(movies)
+            user_data["watchlist"].remove(movies)
+    return user_data
+        
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    length = len(user_data["watched"])
+    if length==0:
+        return 0
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    avg = sum/ length
+    return avg            
+
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    # if user_data["watched"]==[]:
+    # length = len(user_data["watched"])
+    # if length==0:
+    #     return None
+    if not user_data["watched"]:
+        return None 
+    for movie in user_data["watched"]:
+        genre = movie["genre"] 
+        if genre not in genre_count:
+            genre_count[genre] = 1
+        else:
+            genre_count[genre] += 1
+    sorted_genre = sorted(genre_count.items(), key = lambda kv:kv[1])
+    return sorted_genre[-1][0]
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def friends_list(user_data):
+    friends_movie_list = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_movie_list:
+                friends_movie_list.append(movie)
+    return friends_movie_list
+
+def users_list(user_data):
+    users_movie_list= []
+    for movie in user_data["watched"]:
+        if movie not in users_movie_list:
+            users_movie_list.append(movie)
+    return users_movie_list
+
+def get_unique_watched(user_data):
+    user_unique_list=[]
+    friends_movies = friends_list(user_data)
+    users_movies = users_list(user_data)
+   
+    for movie in users_movies:
+        if movie not in friends_movies:
+            user_unique_list.append(movie)
+    return user_unique_list
+    # print(unique_list == friends_movies - users_movies)
+    # return unique_list == friends_movies - users_movies 
+    # user_unique_list == list(set(friends_movies)-set(users_movies))     
+    # return user_unique_list
+
+def get_friends_unique_watched(user_data):
+    friend_unique_list=[]
+    friends_movies = friends_list(user_data)
+    users_movies = users_list(user_data)
+
+    for movie in friends_movies:
+        if movie not in users_movies:
+            friend_unique_list.append(movie)
+    return friend_unique_list
+#     # print(unique_list == friends_movies - users_movies)
+#     # return unique_list == friends_movies - users_movies 
+#     friend_unique_list == list(set(friends_movies)-set(users_movies))     

```

Comment:
See my comment above about sets. Also remember that `==` is checking equality so line 128 is checking that friend_unique_list is equal to what's on the right side. Were you meaning to use the assignment operator `=` ?